pkgs/applications/editors/standardnotes/default.nix

```
@@ -1,45 +1,38 @@
-{ lib, stdenv, fetchurl, dpkg, makeWrapper, electron, libsecret
-, desktop-file-utils , callPackage }:
+{ callPackage, lib, stdenv, appimageTools, autoPatchelfHook, desktop-file-utils
+, fetchurl, libsecret  }:
 
 let
-
   srcjson = builtins.fromJSON (builtins.readFile ./src.json);
-
-  throwSystem = throw "Unsupported system: ${stdenv.hostPlatform.system}";
-
-in
-
-stdenv.mkDerivation rec {
-
+  version = srcjson.version;
   pname = "standardnotes";
+  name = "${pname}-${version}";
+  throwSystem = throw "Unsupported system: ${stdenv.hostPlatform.system}";
 
-  src = fetchurl (srcjson.deb.${stdenv.hostPlatform.system} or throwSystem);
-
-  inherit (srcjson) version;
-
-  dontConfigure = true;
-
-  dontBuild = true;
+  src = fetchurl (srcjson.appimage.${stdenv.hostPlatform.system} or throwSystem);
 
-  nativeBuildInputs = [ makeWrapper dpkg desktop-file-utils ];
+  appimageContents = appimageTools.extract {
+    inherit name src;
+  };
 
-  unpackPhase = "dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner";
+  nativeBuildInputs = [ autoPatchelfHook desktop-file-utils ];
 
-  installPhase = ''
-    runHook preInstall
+in appimageTools.wrapType2 rec {
+  inherit name src;
 
-    mkdir -p $out/bin $out/share/standardnotes
-    cp -R usr/share/{applications,icons} $out/share
-    cp -R opt/Standard\ Notes/resources/app.asar $out/share/standardnotes/
+  extraPkgs = pkgs: with pkgs; [
+    libsecret
+  ];
 
-    makeWrapper ${electron}/bin/electron $out/bin/standardnotes \
-      --add-flags $out/share/standardnotes/app.asar \
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libsecret stdenv.cc.cc.lib ]}
+  extraInstallCommands = ''
+    # directory in /nix/store so readonly
+    cd $out
+    chmod -R +w $out
```

```suggestion
    chmod -R +w $out
```
we don't need to cd here